.ci/docker/requirements-ci.txt

```diff
@@ -298,7 +298,7 @@ pytest-cpp==2.3.0
 #Pinned versions: 2.3.0
 #test that import:
 
-z3-solver==4.12.2.0
```

Comment:
Note from the OH: we upgrade z3-solver because it has a fix for s390x (@AlekseiNikiforovIBM  to provide more details)